lib/wasix/src/state/env.rs

```diff
@@ -1277,7 +1277,7 @@ impl WasiEnv {
         }
 
         // If this is the main thread then also close all the files
-        if self.thread.is_main() {
+        if self.thread.is_main() || matches!(exit_code, Some(e) if !e.is_success()) {
```

Comment:
We have to make sure this code path isn't hit for subprocesses.
First with manual testing, but I think we also need both negative and positive tests here.

Reply:
I've already tested everything I can think of with bash. This code path _is_ hit for subprocesses, but only terminates the subprocess, not the parent process.